File: AccuPay/Domain/Paystubs/PayFrequency.vb
Option Strict On

Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema

Namespace Global.AccuPay.Entity

    <Table("payfrequency")>
    Public Class PayFrequency

        Public Const WeeklyType As String = "WEEKLY"

        Public Const SemiMonthlyType As String = "SEMI-MONTHLY"

        <Key>
        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property RowID As Integer?

        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property Created As Date

        Public Property CreatedBy As Integer?

        <DatabaseGenerated(DatabaseGeneratedOption.Computed)>
        Public Property LastUpd As Date?

        Public Property LastUpdBy As Integer?

        <Column("PayFrequencyType")>
        Public Property Type As String

        Public ReadOnly Property IsWeekly As Boolean
            Get
                Return Type.ToUpper() = WeeklyType
            End Get
        End Property

        Public ReadOnly Property IsSemiMonthly As Boolean
            Get
                Return Type.ToUpper() = SemiMonthlyType
            End Get
        End Property

    End Class

End Namespace

File: AccuPay.Core/Entities/PayFrequency.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AccuPay.Core.Entities
{
    [Table("payfrequency")]
    public class PayFrequency
    {
        public const string WeeklyType = "WEEKLY";

        public const string SemiMonthlyType = "SEMI-MONTHLY";

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int? RowID { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public DateTime Created { get; set; }

        public int? CreatedBy { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime? LastUpd { get; set; }

        public int? LastUpdBy { get; set; }

        [Column("PayFrequencyType")]
        public string Type { get; set; }

        public bool IsWeekly => Type.ToUpper() == WeeklyType;

        public bool IsSemiMonthly => Type.ToUpper() == SemiMonthlyType;
    }
}